### PR TITLE
Allow `add_ip_pool` to write the IP_IN_IP etcd key when `/calico` empty

### PR DESCRIPTION
**Pull request by alexhersh**
- add_ip_pool currently performs a read before it decides to write to the global IP_IN_IP key
- In ipip deployments, we recommend that users call `pool add` before `node`, meaning that `pool add` needs to be able to write to an empty etcd

### CODE REVIEW

File: calico_containers/pycalico/datastore.py [hunk 1]
```diff
@@ -341,8 +341,11 @@ def add_ip_pool(self, version, pool):
         # If IP in IP is enabled on the pool, ensure that it is enabled
         # globally.
         if pool.ipip:
-            result = self.etcd_client.read(IP_IN_IP_PATH)
-            if result.value != IP_IN_IP_ENABLED:
+            try:
+                result = self.etcd_client.read(IP_IN_IP_PATH)
+            except EtcdException:
```
Comment (caseydavenport) [inline calico_containers/pycalico/datastore.py]: Is this the correct Exception to be catching here?  There are a couple of failure cases here at least.

1) The key does not exist yet in `etcd` (this is the one we care about)
2) We fail to connect to `etcd` at all (we don't want to catch this)

File: calico_containers/pycalico/datastore.py [hunk 2]
```diff
@@ -341,8 +341,11 @@ def add_ip_pool(self, version, pool):
         # If IP in IP is enabled on the pool, ensure that it is enabled
         # globally.
         if pool.ipip:
-            result = self.etcd_client.read(IP_IN_IP_PATH)
-            if result.value != IP_IN_IP_ENABLED:
+            try:
```
Comment (caseydavenport) [inline calico_containers/pycalico/datastore.py:362]: Could you add a comment describing the purpose of this try/except, and why we perform a read first at all?

**Comment by caseydavenport** [comment]
I suspect this needs a UT.

**Comment by spikecurtis** [comment]
LGTM.  Can you squash commits?

**Pull request merged**